### PR TITLE
Code Optimization for the Elixir SDK

### PR DESCRIPTION
**Pull request by imahmedismail**
I've added 2 most important packages i.e. credo and dialyxir - which pointed out many of the issues in the code. I've fixed most of them but few are still remaining, which can either be done by me or maybe other developers working on it. Alongwith that I've also added the module, function docs to Utils module.

**Note:** This is an open-source contribution

### CODE REVIEW

**Comment by aliaamuqaddas** [comment]
LGTM

File: lib/kinde_sdk/kinde_client_sdk.ex [hunk 1]
```diff
@@ -250,22 +221,8 @@ defmodule KindeClientSDK do
...
-            domain: binary,
-            grant_type: any,
-            logout_endpoint: <<_::56, _::_*8>>,
-            logout_redirect_uri: any,
-            redirect_uri: binary,
-            scopes: any,
-            token_endpoint: <<_::64, _::_*8>>
-          },
-          map
+          map(),
```
Comment (femz12) [inline lib/kinde_sdk/kinde_client_sdk.ex:224]: Looks good but to be more precise with the spec like it was before

we can use

`KindeClientSDK.t()`

and declate it

`  @type t :: %KindeClientSDK{
    additional_params: map,
    auth_status: atom,
    authorization_endpoint: binary,
    cache_pid: pid,
    client_id: any,
    client_secret: any,
    domain: binary,
    grant_type: any,
    logout_endpoint: binary,
    logout_redirect_uri: any,
    redirect_uri: binary,
    scopes: any,
    token_endpoint: binary
  }`

**Review by femz12 — APPROVED** [review]
Thanks for the contribution

File: lib/kinde_sdk/kinde_client_sdk.ex (hunk 1 above)
Comment (onderay) [reply lib/kinde_sdk/kinde_client_sdk.ex:224]: Thanks @imahmedismail, once the changes have been made. We will merge

**Comment by DanielRivers** [comment]
Once the changes have been made @imahmedismail Could you resolve the conflicts, I merged other PR

File: lib/kinde_sdk/kinde_client_sdk.ex (hunk 1 above)
Comment (imahmedismail) [reply lib/kinde_sdk/kinde_client_sdk.ex]: Done with the changes
cc: @onderay @femz12

**Review by Hammadkhan0034 — APPROVED** [review]

File: lib/kinde_sdk/kinde_client_sdk.ex [hunk 2]
```diff
@@ -426,6 +354,26 @@ defmodule KindeClientSDK do
     throw("Please provide correct grant_type")
   end
 
+  defp fetch_token_with_expiring_timestamps(conn, expiring_timestamp, form_params, client) do
```
Comment (Hammadkhan0034) [inline lib/kinde_sdk/kinde_client_sdk.ex:357]: Documentation can be added for private functions just like for public functions. This allows you to clearly describe the purpose and usage of private functions, making the codebase easier to understand and maintain for anyone working with it.

**Comment by Hammadkhan0034** [comment]
All changes are as per description given in explanation of PR.

**Pull request merged**